auth/custom-credentials/aws/README.md

```diff
@@ -0,0 +1,131 @@
+
+This sample demonstrates how to use **Boto3** (the AWS SDK for Python) as a custom supplier to bridge AWS credentials—from sources like EKS IRSA, ECS, or Fargate—to Google Cloud Workload Identity.
+
+## Running Locally
+
+To run the sample on your local system, you need to install the dependencies and configure your AWS and GCP credentials as environment variables.
+
+### 1. Install Dependencies
+
+Ensure you have Python installed, then install the required libraries:
+
+```bash
+pip install -r requirements.txt
+```
+
+### 2. Set Environment Variables
+
+```bash
+export AWS_ACCESS_KEY_ID="YOUR_AWS_ACCESS_KEY_ID"
+export AWS_SECRET_ACCESS_KEY="YOUR_AWS_SECRET_ACCESS_KEY"
+export AWS_REGION="YOUR_AWS_REGION" # e.g., us-east-1
+export GCP_WORKLOAD_AUDIENCE="YOUR_GCP_WORKLOAD_AUDIENCE"
+export GCS_BUCKET_NAME="YOUR_GCS_BUCKET_NAME"
+
+# Optional: If you want to use service account impersonation
+export GCP_SERVICE_ACCOUNT_IMPERSONATION_URL="YOUR_GCP_SERVICE_ACCOUNT_IMPERSONATION_URL"
+```
+
+### 3. Run the Script
+
+```bash
+python3 snippets.py
+```
+
+## Running in a Containerized Environment (EKS)
+
+This section provides a brief overview of how to run the sample in an Amazon EKS cluster.
+
+### 1. EKS Cluster Setup
+
+First, you need an EKS cluster. You can create one using `eksctl` or the AWS Management Console. For detailed instructions, refer to the [Amazon EKS documentation](https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html).
+
+### 2. Configure IAM Roles for Service Accounts (IRSA)
+
+IRSA allows you to associate an IAM role with a Kubernetes service account. This provides a secure way for your pods to access AWS services without hardcoding long-lived credentials.
+
+- Create an IAM OIDC provider for your cluster.
+- Create an IAM role and policy that grants the necessary AWS permissions.
+- Associate the IAM role with a Kubernetes service account.
+
+For detailed steps, see the [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) documentation.
+
+### 3. Configure GCP to Trust the AWS Role
+
+You need to configure your GCP project to trust the AWS IAM role you created. This is done by creating a Workload Identity Pool and Provider in GCP.
+
+- Create a Workload Identity Pool.
+- Create a Workload Identity Provider that trusts the AWS role ARN.
+- Grant the GCP service account the necessary permissions.
+
+### 4. Containerize and Package the Application
+
+Create a `Dockerfile` for the Python application and push the image to a container registry (e.g., Amazon ECR) that your EKS cluster can access.
+
+**Dockerfile**
```

Comment:
instead of embedding the Dockerfile in the documentation, you can just add the Dockerfile and refer to it. that makes it easier to keep updated.